packages/block-library/src/query-pagination-previous/index.php

```diff
@@ -17,31 +17,63 @@
 function render_block_core_query_pagination_previous( $attributes, $content, $block ) {
 	$page_key = isset( $block->context['queryId'] ) ? 'query-' . $block->context['queryId'] . '-page' : 'query-page';
 	$page     = empty( $_GET[ $page_key ] ) ? 1 : (int) $_GET[ $page_key ];
+	$paged    = ( get_query_var( 'paged' ) ) ? get_query_var( 'paged' ) : 1;
+	$max_page = isset( $block->context['query']['pages'] ) ? (int) $block->context['query']['pages'] : 0;
+
+	$wrapper_attributes        = get_block_wrapper_attributes();
+	$hidden_wrapper_attributes = get_block_wrapper_attributes( array( 'aria-hidden' => 'true' ) );
+	$default_label             = __( 'Previous Page' );
+	$label                     = isset( $attributes['label'] ) && ! empty( $attributes['label'] ) ? $attributes['label'] : $default_label;
+	$pagination_arrow          = get_query_pagination_arrow( $block, false );
+	$content                   = '';
 
-	$wrapper_attributes = get_block_wrapper_attributes();
-	$default_label      = __( 'Previous Page' );
-	$label              = isset( $attributes['label'] ) && ! empty( $attributes['label'] ) ? $attributes['label'] : $default_label;
-	$pagination_arrow   = get_query_pagination_arrow( $block, false );
 	if ( $pagination_arrow ) {
 		$label = $pagination_arrow . $label;
 	}
-	$content = '';
 	// Check if the pagination is for Query that inherits the global context
 	// and handle appropriately.
 	if ( isset( $block->context['query']['inherit'] ) && $block->context['query']['inherit'] ) {
+		global $wp_query;
+		$max_page               = ! $max_page || $max_page > $wp_query->max_num_pages ? $wp_query->max_num_pages : $max_page;
 		$filter_link_attributes = function() use ( $wrapper_attributes ) {
 			return $wrapper_attributes;
 		};
+
 		add_filter( 'previous_posts_link_attributes', $filter_link_attributes );
-		$content = get_previous_posts_link( $label );
+
+		// If there are pages to paginate...
+		if ( 1 < $max_page ) {
+			if ( 1 !== $paged ) { // ... and we are NOT in the first one.
+				$content = get_previous_posts_link( $label );
+			} else { // ... and we are in the first one.
+				$content = sprintf(
+					'<span %1$s>%2$s</span>',
+					$hidden_wrapper_attributes,
+					$label
+				);
+			}
+		}
 		remove_filter( 'previous_posts_link_attributes', $filter_link_attributes );
-	} elseif ( 1 !== $page ) {
-		$content = sprintf(
-			'<a href="%1$s" %2$s>%3$s</a>',
-			esc_url( add_query_arg( $page_key, $page - 1 ) ),
-			$wrapper_attributes,
-			$label
-		);
+	} elseif ( ! $max_page || $max_page > $page ) {
+		$custom_query  = new WP_Query( build_query_vars_from_query_block( $block, $page ) );
+		$max_num_pages = $custom_query->max_num_pages ? $custom_query->max_num_pages : 1;
```

Comment:
Why do we need this `max_num_pages` logic? Previously it would enter this `if` when it was a `custom` query and it wasn't on the first page. So wouldn't a check about `$page === 1` be enough? Am I missing something?

Reply:
This is necessary because if the query has no results  `$custom_query->max_num_pages` **equals to zero**. So the logic to know if there are multiple pages `1 < $max_num_pages` doesn't work.

Reply:
I'm sorry but I still don't get it. In `Query Pagination Previous` we only care about whether we are on the first page or not, no? If we are on the first page, we don't have a link to show, but if we are not we always have a link. This is different from `Query Pagination Next` where we have to recreate the query run from `Query Loop` to know if we have a link for the next posts.

Reply:
Nope, we don't care only about if we are on the first page or not. We need to know if there are pages to paginate. If there are not pages to paginate should not render anything.